ext/intl/formatter/formatter_format.c

```diff
@@ -53,30 +53,55 @@ PHP_FUNCTION( numfmt_format )
 			case IS_DOUBLE:
 				type = FORMAT_TYPE_DOUBLE;
 				break;
-			EMPTY_SWITCH_DEFAULT_CASE();
+			case IS_STRING:
+				type = FORMAT_TYPE_DECIMAL;
+				break;
+			default:
+				zend_argument_type_error(1, "must be of type int|float|string, %s given", zend_zval_type_name(number));
```

Comment:
This will fail with a Stringable object. Also, booleans are now not accepted in weak mode (they behave like ``int``) but that is more of a whatever, so just add ``IS_OBJECT`` at that same level as ``IS_STRING`` and the type error caused by the conversion to string will handle the bailing out.

Reply:
Alright, Stringable objects should now work with `TYPE_DEFAULT` / `TYPE_DECIMAL` (and I added it to the test) – but not with `TYPE_INT64`: it looks like `zval_try_get_long()` doesn’t support Stringable objects (whereas `try_convert_to_string()` does).

Reply:
This is actually expected behaviour, stringable objects do not coerce to int, float, or bool.

Reply:
Alright :+1: